altairexpress/scatter.py

```diff
@@ -1,14 +1,76 @@
 
     Arguements:
 
-    df - (dataframe) Dataframe 
+    data - (dataframe) Dataframe 
     xval - (String) variable/column name 
     yval - (String) variable/column name
     x_logtransform - (Boolean) determines whether a natural log transformation occurs on the x-axis
     y_logtransform - (Boolean) determines whether a natural log transformation occurs on the y-axis
     '''
-    return
\ No newline at end of file
+    # create a copy of the dataframe so that original dataframe remains unchanged
+    df = data.copy(deep = True)
+   
+    # Check to ensure variable types are appropriate
+    assert isinstance(df, pd.DataFrame), "TypeError: Data must be entered as a pandas dataframe."
+    assert isinstance(xval, str), "Wrong type! X-axis variable must be entered and be of type String."
+    assert isinstance(yval, str), "Wrong type! X-axis variable must be entered and be of type String."
+    assert isinstance(x_transform, bool), "TypeError: x_transform must be of type boolean."
+    assert isinstance(y_transform, bool), "TypeError: y_transform must be of type boolean."
+    
+    # Ensure variable exists in the dataframe
+    if xval not in df.columns or yval not in df.columns:
+        raise ValueError("Variable name not found in input dataframe. Double-check spelling!")
+    
+    # Ensure variable is numeric
+    assert pd.api.types.is_numeric_dtype(df[xval]), "Your x-variable needs to be numeric. Double-check dtype and coerce to numeric if necessary."
+    assert pd.api.types.is_numeric_dtype(df[yval]), "Your y-variable needs to be numeric. Double-check dtype and coerce to numeric if necessary."  
+    
+
+    # Toggle a log transformation on the x-axis
+    if x_transform == True:
+        if any(df[xval] < 0):
+            warnings.warn("You have negative values in your x variable, which is incompatible with np.log.")
+        df[xval] = np.log(df[xval])
+
+            
+    # Toggle a log transformation on the y-axis
+    if y_transform == True:
+        if any(df[yval] < 0):
+            warnings.warn("You have negative values in your y variable, which is incompatible with np.log.")
+        df[yval] = np.log(df[yval])
+    
+    # Update scale bounds of the plots
+    x_scale = alt.Scale(domain = (float(df[xval].min()), float(df[xval].max())))
+    y_scale = alt.Scale(domain = (float(df[yval].min()), float(df[yval].max())))
+
+    # Create base scatterplot
+    scatter = alt.Chart(df).mark_circle(size=90, opacity = 0.3).encode(
+                alt.X(xval, scale = x_scale),
+                alt.Y(yval, scale = y_scale)
+            ).properties(width=300, height=200)
+
+    # Create x-histogram
+    x_hist = alt.Chart(df).mark_bar(opacity = 0.3).encode(
+            alt.X(xval, bin=alt.Bin(extent = x_scale.domain), title = ""),
+            alt.Y('count()', stack = None)               
+        ).properties(title='{0} distribution'.format(xval),
+                        width=300, height=100)
```

Comment:
An additional feature here could allow the user to supply a string for the axis labels and title of their choice. Then you could insert them here in the chart to allow a custom labeling.